File: CS/DropDownRemoteData/Controllers/OrdersController.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Threading.Tasks;
using System.Web.Http;
using System.Web.Http.Description;
using DropDownRemoteData.Models;
using DevExtreme.AspNet.Data;
using DevExtreme.AspNet.Data.Helpers;

namespace DropDownRemoteData.Controllers
{
    public class OrdersController : ApiController
    {
        private NorthwindEntities db = new NorthwindEntities();

        // GET: api/Orders
        [HttpGet]
        public HttpResponseMessage GetOrders()
        {
            DataSourceLoadOptions parameters = GetLoadParams();
            var result = DataSourceLoader.Load(db.Orders, parameters);
            return Request.CreateResponse(result);
        }

        private DataSourceLoadOptions GetLoadParams()
        {
            DataSourceLoadOptions parameters = new DataSourceLoadOptions();
            var requestValues = Request.GetQueryNameValuePairs();
            DataSourceLoadOptionsParser.Parse(parameters, (key) =>
            {
                var items = requestValues.Where(i => i.Key == key);
                if (items.Count() > 0)
                    return items.FirstOrDefault().Value;
                else
                    return null;
            });
            parameters.PrimaryKey = new string[] { "OrderID" };
            return parameters;
        }

        // GET: api/Orders/5
        [ResponseType(typeof(Order))]
        public async Task<IHttpActionResult> GetOrder(int id)
        {
            Order order = await db.Orders.FindAsync(id);
            if (order == null)
            {
                return NotFound();
            }

            return Ok(order);
        }
        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                db.Dispose();
            }
            base.Dispose(disposing);
        }

        private bool OrderExists(int id)
        {
            return db.Orders.Count(e => e.OrderID == id) > 0;
        }
    }
}

File: VB/DropDownRemoteData/Controllers/OrdersController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Data.Entity.Infrastructure
Imports System.Linq
Imports System.Net
Imports System.Net.Http
Imports System.Threading.Tasks
Imports System.Web.Http
Imports System.Web.Http.Description
Imports DropDownRemoteData.Models
Imports DevExtreme.AspNet.Data
Imports DevExtreme.AspNet.Data.Helpers

Namespace DropDownRemoteData.Controllers
	Public Class OrdersController
		Inherits ApiController
		Private db As New NorthwindEntities()

		' GET: api/Orders
		<HttpGet> _
		Public Function GetOrders() As HttpResponseMessage
			Dim parameters As DataSourceLoadOptions = GetLoadParams()
			Dim result = DataSourceLoader.Load(db.Orders, parameters)
			Return Request.CreateResponse(result)
		End Function

		Private Function GetLoadParams() As DataSourceLoadOptions
			Dim parameters As New DataSourceLoadOptions()
			Dim requestValues = Request.GetQueryNameValuePairs()
            DataSourceLoadOptionsParser.Parse(parameters, Function(key) DelegateHandler(key, requestValues))
            parameters.PrimaryKey = New String() {"OrderID"}
            Return parameters
        End Function

        'INSTANT VB TODO TASK: The return type of this anonymous method could not be determined by Instant VB:
        Private Function DelegateHandler(ByVal key As Object, ByVal requestValues As IEnumerable(Of KeyValuePair(Of String, String))) As String
            Dim items = requestValues.Where(Function(i) i.Key Is key)
            If items.Count() > 0 Then
                Return items.FirstOrDefault().Value
            Else
                Return Nothing
            End If
        End Function

		' GET: api/Orders/5
		<ResponseType(GetType(Order))> _
		Public async Function GetOrder(ByVal id As Integer) As Task(Of IHttpActionResult)
			Dim order As Order = await db.Orders.FindAsync(id)
			If order Is Nothing Then
				Return NotFound()
			End If

			Return Ok(order)
		End Function
		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				db.Dispose()
			End If
			MyBase.Dispose(disposing)
		End Sub

		Private Function OrderExists(ByVal id As Integer) As Boolean
			Return db.Orders.Count(Function(e) e.OrderID = id) > 0
		End Function
	End Class
End Namespace